packages/client/src/components/block-defaults/stepper-block/StepperBlock.tsx

```diff
@@ -1,31 +1,44 @@
-import React from 'react';
-import { useBlock, useDebounce } from '@/hooks';
+import { useBlock } from '@/hooks';
 import { BlockComponent, BlockDef } from '@/stores';
 import { observer } from 'mobx-react-lite';
-import { Step, StepLabel, Stepper } from '@mui/material';
+import { Step, StepLabel, Stepper, Typography } from '@mui/material';
+
+interface Title {
+    name: string;
+    id?: number;
+}
 
 export interface StepperBlockDef extends BlockDef<'stepper'> {
     widget: 'stepper';
     data: {
-        steps: [];
-        activeStep: 0;
+        active: number;
+        steps: Title[];
     };
 }
 
 export const StepperBlock: BlockComponent = observer(({ id }) => {
-    const { attrs, data, setData, listeners } = useBlock<StepperBlockDef>(id);
-
-    const { steps, activeStep } = data;
-
+    const { attrs, data } = useBlock<StepperBlockDef>(id);
+    const { active } = data;
     return (
-        <Stepper {...attrs}>
-            {steps.map((label, i) => {
-                return (
-                    <Step key={`${id}--${label}--${i}`}>
-                        <StepLabel>{label}</StepLabel>
-                    </Step>
-                );
-            })}
-        </Stepper>
+        <>
+            <Stepper {...attrs} activeStep={active}>
+                {data.steps?.map((obj, i) => {
+                    return (
+                        <Step key={`${id}--${obj.name}--${i}`}>
+                            <StepLabel>{obj.name}</StepLabel>
+                        </Step>
+                    );
+                })}
+            </Stepper>
+            {Number(active) >= data.steps?.length ? (
```

Comment:
This whole block of showing the content, should not be apart of this block. (Content will be rendered by another block, where we will use a conditional) based on the activeStep)

Lines: 33-41